isofit/utils/analytical_line.py

```diff
@@ -30,13 +30,256 @@
+    update_config_for_surface,
+)
+
+
+@ray.remote(num_cpus=1)
+class Worker(object):
+    def __init__(
+        self,
+        config: configs.Config,
+        pixel_index: dict,
+        full_statevector: list,
+        full_idx_surface: np.array,
+        full_idx_RT: np.array,
+        full_idx_surf_rfl: np.array,
+        output_files: dict,
+        output_shape: tuple,
+        loglevel: str,
+        logfile: str,
+        subs_state_file: str = None,
+    ):
+        """
+        Worker class to help run a subset of spectra.
+
+        Args:
+            fm: isofit forward_model
+            loglevel: output logging level
+            logfile: output logging file
+        """
+        logging.basicConfig(
+            format="%(levelname)s:%(asctime)s ||| %(message)s",
+            level=loglevel,
+            filename=logfile,
+            datefmt="%Y-%m-%d,%H:%M:%S",
+        )
+        self.config = config
+        self.pixel_index = pixel_index
+
+        self.esd = IO.load_esd(IO.earth_sun_distance_path)
+
+        self.full_statevector = full_statevector
+        self.full_idx_surface = full_idx_surface
+        self.full_idx_RT = full_idx_RT
+        self.full_idx_surf_rfl = full_idx_surf_rfl
+
+        self.winidx = retrieve_winidx(self.config)
+
+        self.output_shape = output_shape
+        self.rfl_output = output_files["rfl_output"]
+        self.unc_output = output_files["unc_output"]
+
+        self.completed_spectra = 0
+        self.hash_table = OrderedDict()
+        self.hash_size = 500
+
+        if config.input.radiometry_correction_file is not None:
+            self.radiance_correction, wl = load_spectrum(
+                config.input.radiometry_correction_file
+            )
+        else:
+            self.radiance_correction = None
+
+    def run_chunks(self, input_arg: tuple, fill_value: float = -9999.0) -> None:
+        """
+        TODO: Description
+        """
+        # Unpack arguments
+        start_line, stop_line, input_chunk = input_arg
+
+        rdn = input_chunk["rdn_file"]
+        loc = input_chunk["loc_file"]
+        obs = input_chunk["obs_file"]
+        rt_state = input_chunk["rt_state"]
+
+        # Set up outputs
+        output_state = (
+            np.zeros(
+                (
+                    stop_line - start_line,
+                    rt_state.shape[1],
+                    len(self.full_idx_surface),
+                )
+            )
+            + fill_value
+        )
+        output_state_unc = (
+            np.zeros(
+                (
+                    stop_line - start_line,
+                    rt_state.shape[1],
+                    len(self.full_idx_surface),
+                )
+            )
+            + fill_value
+        )
+
+        # Index chunk
+        state_indexes = chunk_surface_spectra(
+            start_line, stop_line, rdn.shape[1], self.pixel_index
+        )
+
+        for surface_class_str, class_idx_pairs in state_indexes.items():
```

Comment:
The strategy here is similar to Isofit.run in that difference forward-inversions are completed serially. The order of multiprocessing on chunks should be unchanged. Each actor iterates through row-column pairs and a chunk is written via write_bil_chunk to the output file.